docs: add  `ngtsc` to glossary

ajitsinghkaler opened the pull request:
ngtsc angular compiler was not mentioned in the glossary. glossary sould contain the relevant terms in angular which are hard to get added ngtsc to glossary

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)


## PR Type
What kind of change does this PR introduce?

<!-- Please check the one that applies to this PR using "x". -->

- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:


## What is the current behavior?
<!-- Please describe the current behavior that you are modifying, or link to a relevant issue. -->
No mention of ngtsc inglossary


Issue Number: N/A


## What is the new behavior?
 mention of ngtsc in glossary


## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No


<!-- If this PR contains a breaking change, please describe the impact and migration path for existing applications below. -->


## Other information

IgorMinar reviewed (commented):
I'm not sure if we want to make ngtsc part of our public vocabulary.

ngtsc is an implementation detail and once all traces of the view engine compiler are removed, ngc will be exactly equal to ngtsc. In other words ngc is the proper public name for the compiler because it matches the binary developers can call on the command line.

In other words ngtsc is just an implementation detail and most developers shouldn't worry about it. In fact it might only confuse them.

@aikidave what do you think?

aikidave commented:
I don't think it does any harm to add this entry to the glossary, but it might be better to just say something like:

> ngtsc: See ngc.

That way, if users come across the term `ngtc` and look it up, they're redirected accordingly. Thoughts?

ajitsinghkaler commented:
@IgorMinar should I dd it in the way @aikidave suggest or should I code this pr

IgorMinar commented:
What Dave suggested works for me.

Maybe we could say that: the internal name for the current generation of the Angular compiler. See ngc.

ajitsinghkaler commented:
@IgorMinar @aikidave we don't even have ngc in documentation we have ngcc but no mention of ngc

ajitsinghkaler commented:
@IgorMinar comments

ajitsinghkaler commented:
@IgorMinar any comments, thoughts???

IgorMinar commented on aio/content/guide/glossary.md:
```diff
@@ -620,6 +620,13 @@ The [npm package manager](https://docs.npmjs.com/getting-started/what-is-npm) is
 
 Learn more about how Angular uses [Npm Packages](guide/npm-packages).
 
+{@ ngtsc}
+
+## ngtsc
+
+`ngtsc` is an Typescript-to-Javascript transpiler that reifies Angular decorators into static properties.
+It is a minimal wrapper around `tsc` which includes a set of Angular transforms to convert angular code into javascript.
+
```
```suggestion
{@ ngc}

## ngc

`ngc` is a Typescript-to-Javascript transpiler that processes Angular decorators, metadata, and templates and emits JavaScript code.
The most recent implementation is internally refered to as `ngtsc` because it's a minimalistic wrapper around the TypeScript compiler `tsc` that adds a transform for processing Angular code.
```

IgorMinar reviewed (changes requested):
@ajitsinghkaler how about this:

IgorMinar reviewed (dismissed):
Reviewed-for: public-api

ajitsinghkaler committed:
docs: add ngc to glossary

ngc angular compiler was not mentioned in the glossary.
Glossary should contain the relevant terms in angular
which are hard to get. So, added a small defination of
ngc to the glossary

jelbourn reviewed (approved):
LGTM

Reviewed-for: public-api

pkozlowski-opensource reviewed (approved):
LGTM

Reviewed-for: public-api

petebacondarwin commented on aio/content/guide/glossary.md:
```diff
@@ -627,6 +627,11 @@ The [npm package manager](https://docs.npmjs.com/getting-started/what-is-npm) is
 
+{@ ngc}
+## ngc
+`ngc` is a Typescript-to-Javascript transpiler that processes Angular decorators, metadata, and templates and emits JavaScript code.
```
```suggestion
`ngc` is a Typescript-to-Javascript transpiler that processes Angular decorators, metadata and templates, and emits JavaScript code.
```

petebacondarwin replied:
If we are using an Oxford comma it ought to be at the end of the list of related items.

ajitsinghkaler commented:
@petebacondarwin made the changes

petebacondarwin reviewed (approved):
LGTM

petebacondarwin reviewed (approved):
Reviewed-for: public-api

ajitsinghkaler commented:
@petebacondarwin this has been approved for a long time can we get this merged

petebacondarwin commented:
Merge-assistance: the PullApprove is wrong. Igor's review was dismissed, so should not be blocking approval.
